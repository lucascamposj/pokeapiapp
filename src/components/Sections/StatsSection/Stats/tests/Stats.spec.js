import React from 'react';
import {shallow} from 'enzyme';
import Stat from '../../Stats';

describe('Stats', () => {
    describe('Rendering', () => {
        it('should match to snapshot with effort > 0', () => {
            const stat =
              {
                stat: {
                  name: "attack",
                },
                effort: 1,
                base_stat: 450
              }
            const component = shallow(
              <Stat
                stat={stat.stat}
                effort={stat.effort}
                base_stat={stat.base_stat}
              />)
            expect(component).toMatchSnapshot()
        });
        it('should match to snapshot with effort = 0', () => {
          const stat =
            {
              stat: {
                name: "attack",
              },
              effort: 0,
              base_stat: 450
            }
          const component = shallow(
            <Stat
              stat={stat.stat}
              effort={stat.effort}
              base_stat={stat.base_stat}
            />)
          expect(component).toMatchSnapshot()
      });
    });
});
