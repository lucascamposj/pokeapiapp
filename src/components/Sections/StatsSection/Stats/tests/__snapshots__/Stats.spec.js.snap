// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Rendering should match to snapshot with effort = 0 1`] = `
<Styled(View)>
  <Styled(Text)>
    attack
  </Styled(Text)>
  <Styled(View)>
    <Styled(Text)>
      450
    </Styled(Text)>
    <Styled(Text)>
      0
    </Styled(Text)>
  </Styled(View)>
</Styled(View)>
`;

exports[`Stats Rendering should match to snapshot with effort > 0 1`] = `
<Styled(View)>
  <Styled(Text)>
    attack
  </Styled(Text)>
  <Styled(View)>
    <Styled(Text)>
      450
    </Styled(Text)>
    <Styled(Text)>
       (1)
    </Styled(Text)>
  </Styled(View)>
</Styled(View)>
`;
